infinispan-remote/query/src/main/java/org/infinispan/tutorial/simple/remote/query/Person.java

```diff
@@ -1,24 +1,32 @@
 package org.infinispan.tutorial.simple.remote.query;
 
+import org.infinispan.api.annotations.indexing.Decimal;
+import org.infinispan.api.annotations.indexing.Indexed;
+import org.infinispan.api.annotations.indexing.Keyword;
 import org.infinispan.protostream.annotations.ProtoFactory;
 import org.infinispan.protostream.annotations.ProtoField;
 
 /**
  * This class is annotated with the infinispan Protostream support annotations.
  * With this method, you don't need to define a protobuf file and a marshaller for the object.
  */
+@Indexed
 public final class Person {
 
    @ProtoField(number = 1)
+   @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false)
    String firstName;
 
    @ProtoField(number = 2)
+   @Keyword(projectable = true, sortable = true, normalizer = "lowercase", indexNullAs = "unnamed", norms = false)
    String lastName;
 
    @ProtoField(number = 3, defaultValue = "-1")
+   @Decimal
```

Comment:
Just noticed @karesti.
Usually the decimal should be used if a decimal scale different from 0 is present, for instance 2 for a decimal value of a currency field.
In this case you can simply use the `@Basic` annotation.
Same for the string fields that do not require a normalizer or analyzer, you can use `@Basic`.

Reply:
basic means take the field as it is, without no transformation

Reply:
On a numeric field, to use a `@Decimal`with no scale  is equivalent to use a `@Basic`.
As on a string field, to use a `@Keyword` with no normalizer is equivalent to a `@Text` with no analyzer that is equivalent to using a `@Basic`.
In general I would advise to use `@Basic`, if it is possible.